src/main/java/io/cryostat/rules/ScheduledArchiveJob.java

```diff
@@ -47,27 +46,27 @@ class ScheduledArchiveJob implements Job {
     String archiveBucket;
 
     @Override
-    public void execute(JobExecutionContext ctx) throws JobExecutionException {
-        try {
-            var rule = (Rule) ctx.getJobDetail().getJobDataMap().get("rule");
-            var target = (Target) ctx.getJobDetail().getJobDataMap().get("target");
-            var recording = (ActiveRecording) ctx.getJobDetail().getJobDataMap().get("recording");
+    public void execute(JobExecutionContext ctx) {
+        var rule = (Rule) ctx.getJobDetail().getJobDataMap().get("rule");
+        var target = (Target) ctx.getJobDetail().getJobDataMap().get("target");
+        var recording = (ActiveRecording) ctx.getJobDetail().getJobDataMap().get("recording");
 
-            Queue<String> previousRecordings = new ArrayDeque<>(rule.preservedArchives);
+        Queue<String> previousRecordings = new ArrayDeque<>(rule.preservedArchives);
 
-            initPreviousRecordings(target, rule, previousRecordings);
+        initPreviousRecordings(target, rule, previousRecordings);
 
-            while (previousRecordings.size() >= rule.preservedArchives) {
-                pruneArchive(target, previousRecordings, previousRecordings.remove());
-            }
-            performArchival(recording, previousRecordings);
-        } catch (Exception e) {
-            logger.error(e);
-            // TODO: Handle JMX/SSL errors
+        while (previousRecordings.size() >= rule.preservedArchives) {
+            pruneArchive(target, previousRecordings, previousRecordings.remove());
```

Comment:
Also wrap in `QuarkusTransaction.joiningExisting()`?

Reply:
I don't think it's needed here - this only operates on POJOs in memory and on S3 storage, not on any database table.

Reply:
Ahh right! Make sense! I saw the old code with `@Transactional` but didnt look into it! Thanks!

Reply:
Yea, I don't know why that was there to begin with honestly. I'm not sure if it was actually doing anything - it isn't clear to me whether Quartz jobs are considered a "managed bean" where the framework knows how to wrap it in a transaction by the annotation. It seems like using `joiningExisting()` is a better bet anywhere where it's needed and isn't definitely supported by the annotation - so anywhere that isn't an HTTP endpoint or an `@Observes`, `@ConsumeEvent`, or other core framework hook.